src/React/Promise/functions.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace React\Promise;
+
+if (function_exists('React\Promise\resolve')) {
+    return;
```

Comment:
IIRC PHP only supports conditional definitions if they are actually wrapped in conditonals? In other words, has this very statement been tested? :)